libyul/AsmParser.cpp

```diff
@@ -128,21 +128,22 @@ void Parser::fetchSourceLocationFromComment()
 		auto const end = toInt(matchResult[4].str());
 
 		auto const commentLocation = m_scanner->currentCommentLocation();
-		m_locationOverride = SourceLocation{};
+		m_debugDataOverride = DebugData::create();
 		if (!sourceIndex || !start || !end)
 			m_errorReporter.syntaxError(6367_error, commentLocation, "Invalid value in source location mapping. Could not parse location specification.");
 		else if (!((*start < 0 && *end < 0) || (*start >= 0 && *start <= *end)))
 			m_errorReporter.syntaxError(5798_error, commentLocation, "Invalid value in source location mapping. Start offset larger than end offset.");
 		else if (sourceIndex == -1 && (0 <= *start && *start <= *end)) // Use source index -1 to indicate original source.
-			m_locationOverride = SourceLocation{*start, *end, ParserBase::currentLocation().source};
+		{
+			m_debugDataOverride = DebugData::create(SourceLocation{*start, *end, ParserBase::currentLocation().source});
+		}
```

Comment:
```suggestion
			m_debugDataOverride = DebugData::create(SourceLocation{*start, *end, ParserBase::currentLocation().source});
```

Reply:
Actually I think it would be better to keep a single `DebugData` struct and only update its `sourceLocation` member when we encounter `@src`. This will make it easier to add more tags in the future, since it does not reset the other members all the time.